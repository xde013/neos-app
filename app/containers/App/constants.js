/*
 * AppConstants
 */

export const LOAD_NEOS = 'app/HOME/LOAD_NEOS';
export const LOAD_NEOS_SUCCESS = 'app/HOME/LOAD_NEOS_SUCCESS';
export const LOAD_NEOS_ERROR = 'app/HOME/LOAD_NEOS_ERROR';
