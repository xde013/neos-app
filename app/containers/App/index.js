/**
 * App
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route } from 'react-router-dom';
import Container from '@material-ui/core/Container';
import HomePage from 'containers/HomePage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import GlobalStyle from '../../global-styles';
import Header from '../../components/Header';
import Footer from '../../components/Footer';
import AboutPage from '../AboutPage/Loadable';

export default function App() {
  return (
    <Container>
      <Helmet
        titleTemplate="%s - NEOs Nasa"
        defaultTitle="Near Earth Object by Ryan Bourhil"
      >
        <meta
          name="description"
          content="A React.js project showcasing Near Earth Objects"
        />
      </Helmet>
      <Header />

      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route path="/about" component={AboutPage} />
        <Route path="" component={NotFoundPage} />
      </Switch>

      <Footer />
      <GlobalStyle />
    </Container>
  );
}
