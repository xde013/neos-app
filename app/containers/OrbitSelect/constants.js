/*
 *
 * OrbitSelect constants
 *
 */

export const SET_FILTER = 'app/OrbitSelect/SET_FILTER';
