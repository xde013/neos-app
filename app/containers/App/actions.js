/*
 * App Actions
 */

import { LOAD_NEOS, LOAD_NEOS_SUCCESS, LOAD_NEOS_ERROR } from './constants';

export function loadNeos() {
  return {
    type: LOAD_NEOS,
  };
}

export function neosLoaded(neos) {
  return {
    type: LOAD_NEOS_SUCCESS,
    neos,
  };
}

export function neosLoadingError(error) {
  return {
    type: LOAD_NEOS_ERROR,
    error,
  };
}
