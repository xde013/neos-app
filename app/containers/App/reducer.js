/*
 * AppReducer
 *
 */

import produce from 'immer';
import { LOAD_NEOS_SUCCESS, LOAD_NEOS, LOAD_NEOS_ERROR } from './constants';

// The initial state of the App
export const initialState = {
  loading: false,
  error: false,
  userData: {
    neos: false,
  },
};

/* eslint-disable default-case, no-param-reassign */
const appReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_NEOS:
        draft.loading = true;
        draft.error = false;
        break;

      case LOAD_NEOS_SUCCESS:
        draft.userData.neos = action.neos;
        draft.loading = false;
        break;

      case LOAD_NEOS_ERROR:
        draft.error = action.error;
        draft.loading = false;
        break;
    }
  });

export default appReducer;
