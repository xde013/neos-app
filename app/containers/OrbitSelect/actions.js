/*
 *
 * OrbitSelect actions
 *
 */

import { SET_FILTER } from './constants';

export function setFilter(filter) {
  return {
    type: SET_FILTER,
    filter,
  };
}
