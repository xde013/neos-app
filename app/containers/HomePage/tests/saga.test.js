/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_NEOS } from 'containers/App/constants';
import { neosLoaded, neosLoadingError } from 'containers/App/actions';

import neosData, { getNeos } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('getNeos Saga', () => {
  let getNeosGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getNeosGenerator = getNeos();

    const selectDescriptor = getNeosGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the NeosLoaded action if it requests the data successfully', () => {
    const response = {
      near_earth_objects: [
        {
          name: 'First Neo',
        },
        {
          name: 'Second Neo',
        },
      ],
    };
    const putDescriptor = getNeosGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(neosLoaded(response.near_earth_objects)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getNeosGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(neosLoadingError(response)));
  });
});

describe('neosDataSaga Saga', () => {
  const neosDataSaga = neosData();

  it('should start task to watch for LOAD_Neos action', () => {
    const takeLatestDescriptor = neosDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_NEOS, getNeos));
  });
});
