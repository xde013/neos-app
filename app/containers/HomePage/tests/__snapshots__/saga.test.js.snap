// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNeos Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=xLHhDdEiuN5tOg1Z3VnrUkXsBQQv6KHBjbZdIbcX",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getNeos Saga should dispatch the NeosLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://api.nasa.gov/neo/rest/v1/neo/browse?api_key=xLHhDdEiuN5tOg1Z3VnrUkXsBQQv6KHBjbZdIbcX",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
